src/requestQueue.js

```diff
@@ -17,6 +17,12 @@ export default class RequestQueue {
 
```

Comment:
They serve slightly purposes. 

The `dispatchQueue` is an array I use to keep track of requests that are awaiting dispatch. I always dequeue/shift from the head to retrieve the next request to send in `checkQueue()`. This ensures in-order processing. Since we remove requests from this array on dispatch, we can't rely on it to check if a request is pending.

The `activeRequests` map is used to keep track of all pending requests. This includes requests in the `dispatchQueue` and those whose promise has not been fulfilled/rejected. When a new request is added to the `RequestQueue`, I check this map to see if it's a duplicate. If so, I return the existing promise. On fulfillment/rejection of a request, it is removed from 'activeRequests'

The reason both are needed is because `checkQueue` happens asynchronously.